java/src/main/java/org/rocksdb/StringAppendOperator.java

```diff
@@ -18,6 +20,19 @@ public StringAppendOperator(char delim) {
         super(newSharedStringAppendOperator(delim));
     }
 
+    public StringAppendOperator(byte[] delim) {
+        super(newSharedStringAppendTESTOperator(delim));
+    }
+
+    public StringAppendOperator(String delim) {
+        this(delim.getBytes());
+    }
+
+    public StringAppendOperator(String delim, Charset charset) {
+        this(delim.getBytes(charset));
+    }
+
     private native static long newSharedStringAppendOperator(final char delim);
+    private native static long newSharedStringAppendTESTOperator(final byte[] delim);
```

Comment:
Why is this named with "TEST" ?

Reply:
This might be something to discuss. I tried not to change too much in general.

As I understand there are 2 string appending merge operators in c++ at the moment: `StringAppendOperator` and `StringAppendTESTOperator`. The former is the first implementation based on `MergeOperator.FullMerge` before having `MergeOperator.FullMergeV2`. The later is more efficient based on the newer `MergeOperator.FullMergeV2` interface. I changed the newer one in c++. The Java `StringAppendOperator.newSharedStringAppendOperator()` also creates the older one and `newSharedStringAppendTESTOperator` creates the newer one. Basically, now it is in sync with c++ code to some extend but might be a bit confusing in Java code.

I thought about creating another separate Java `MergeOperator` class which is backed by `StringAppendTESTOperator`. It could be called e.g. `StringAppendOperatorWithVariableDelimitor`. I can refactor it if it makes sense.